cypress/e2e/scenario-2.4-new-expert-mode.cy.js

```diff
@@ -0,0 +1,266 @@
+ */
+const clearEnvironment = (nameEnvironment = "lsm-frontend") => {
+  cy.visit("/console/");
+  cy.get('[aria-label="Environment card"]').contains(nameEnvironment).click();
+  cy.url().then((url) => {
+    const location = new URL(url);
+    const id = location.searchParams.get("env");
+
+    cy.request("DELETE", `/api/v1/decommission/${id}`);
+  });
+};
+
+/**
+ * based on the environment id, it will recursively check if a compile is pending.
+ * It will continue the recursion as long as the statusCode is equal to 200
+ *
+ * @param {string} id
+ */
+const checkStatusCompile = (id) => {
+  let statusCodeCompile = 200;
+
+  if (statusCodeCompile === 200) {
+    cy.intercept(`/api/v1/notify/${id}`).as("IsCompiling");
+    // the timeout is necessary to avoid errors.
+    // Cypress doesn't support while loops and this was the only workaround to wait till the statuscode is not 200 anymore.
+    // the default timeout in cypress is 5000, but since we have recursion it goes into timeout for the nested awaits because of the recursion.
+    cy.wait("@IsCompiling", { timeout: 10000 }).then((req) => {
+      statusCodeCompile = req.response.statusCode;
+
+      if (statusCodeCompile === 200) {
+        checkStatusCompile(id);
+      }
+    });
+  }
+};
+
+/**
+ * Will by default execute the force update on the 'lsm-frontend' environment if no argumenst are being passed.
+ * This method can be executed standalone, but is part of the cleanup cycle that is needed before running a scenario.
+ *
+ * @param {string} nameEnvironment
+ */
+const forceUpdateEnvironment = (nameEnvironment = "lsm-frontend") => {
+  cy.visit("/console/");
+  cy.get('[aria-label="Environment card"]').contains(nameEnvironment).click();
+  cy.url().then((url) => {
+    const location = new URL(url);
+    const id = location.searchParams.get("env");
+
+    cy.request({
+      method: "POST",
+      url: `/lsm/v1/exporter/export_service_definition`,
+      headers: { "X-Inmanta-Tid": id },
+      body: { force_update: true },
+    });
+    checkStatusCompile(id);
+  });
+};
+
+if (Cypress.env("edition") === "iso") {
+  describe("Scenario 2.4 Service Catalog - basic-service", () => {
+    before(() => {
+      clearEnvironment();
+      forceUpdateEnvironment();
+    });
+
+    it("2.4.1 Force new state in the instance", () => {
+      // Go from Home page to Service Inventory of Basic-service
+      cy.visit("/console/");
+
+      cy.intercept(
+        "GET",
+        "/lsm/v1/service_inventory/basic-service?include_deployment_progress=True&limit=20&&sort=created_at.desc",
+      ).as("GetServiceInventory");
+
+      cy.get('[aria-label="Environment card"]')
+        .contains("lsm-frontend")
+        .click();
+      cy.get(".pf-v5-c-nav__item").contains("Service Catalog").click();
+      cy.get("#basic-service").contains("Show inventory").click();
+
+      // Make sure the call to get inventory has been executed
+      cy.wait("@GetServiceInventory");
```

Comment:
I forgot the reason behind awaiting the api call.
Shouldn't awaiting for the` ServiceInventory-<State>` would be enough and more in line with e2e approach to imitate how to user would behave?

Reply:
Either solutions are fine, this is what we had in the other tests, so I stayed in line, but it's something we can change in a future iteration to align our E2E tests for example